internal/common/node_module_info.bzl

```diff
@@ -22,13 +22,36 @@ NodeModuleInfo = provider(
     },
 )
 
+NodeModuleSources = provider(
```

Comment:
I suspect this is over complicating things here with two different node_modules providers and a `NODE_MODULE_MARKER` tag. This may be a good opportunity to remove the tag and create a single rule that is used instead of `filegroup` in `generate_build_file.js` so we can avoid the marker and have a single provider.

@kyliau recently added the ng_apf_library() rule which is used instead of `filegroup` when an angular package format is detected. We consolidate this work into a single rule and solve the transitive deps problem as well.